678: Removing abandoned Content Browser module for D11 upgrade.

quicksketch opened the pull request:
Progress towards upgrading to Drupal 11 (https://github.com/MukurtuCMS/Mukurtu-CMS/issues/678).

This removes use of the [Content Browser](https://www.drupal.org/project/content_browser) module, which does not have a Drupal 11 version and the maintainer has not responded to requests.

Previous entity browser when selecting content:
![image](https://github.com/user-attachments/assets/7fad2d1b-6f89-4a00-a5f7-ab7858d50a01)

New entity browser when selecting content:
![image](https://github.com/user-attachments/assets/4a892ca4-875b-44de-9a72-b0484ea6f9e5)

One nice thing is that this eliminates the strange front-end theme displaying in a modal within the back-end pages.

This affects many, many places in Mukurtu where content is referenced, including:

- Block: Featured Content
- Collection: Items in Collection, Sub-Collections, Related Content
- Personal Collection: Items in Collection
- Dictionary Word: Related Content
- Word List: Words
- Digital Heritage: Related Content
- Multipage Item: Pages (where are these added?)
- Person: Related Content
- Related Person (Paragraph): Related Person
- Community: Featured Content

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

quicksketch commented:
This is ready for review now.

@bronzehedwick FYI this removes the "content_browser" display mode. For the most part, this was only used within entity browsers, but it was inexplicably also the display mode used within the `/browse` maps. I switched the maps to use the "browse_grid" display mode, but this might negatively impact the maps in the short term, since I think you just configured maps to display better. Do you think this is okay, or should we make a new "browse_map" display mode to match "browse_grid" and "browse_list"?

quicksketch commented on modules/mukurtu_multipage_items/src/Controller/MultipageItemPageController.php:
```diff
@@ -250,7 +250,7 @@ public function view(NodeInterface $node) {
     return [
       '#theme' => 'multipage_item_book_view',
-      '#pages' => array_map(fn($p) => $view_builder->view($p, 'content_browser'), $pages),
+      '#pages' => array_map(fn($p) => $view_builder->view($p, 'browse_grid'), $pages),
```
```suggestion
      '#pages' => array_map(fn($p) => $view_builder->view($p, 'grid_browse'), $pages),
```

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
@quicksketch that is much cleaner! I'm not running into any specific issues, but do have some questions about possible adjustments. I'm happy to merge this PR as is and address those in a follow-up though!

Also, for Multipage items, when you create one, you can either select existing content to include, or just save as is and THEN there is an option to generate new content.

![Screenshot 2025-03-28 at 8 26 28 AM](https://github.com/user-attachments/assets/50ca048b-1d37-4679-99b0-b97bf37c660b)

![Screenshot 2025-03-28 at 8 27 25 AM](https://github.com/user-attachments/assets/12a006e4-6924-44fb-943b-9ddd6473f4ed)

![Screenshot 2025-03-28 at 8 27 41 AM](https://github.com/user-attachments/assets/2fbb9a35-c3cc-4fb8-b2df-37ff857f29a4)

bronzehedwick commented:
> This is ready for review now.
> 
> @bronzehedwick FYI this removes the "content_browser" display mode. For the most part, this was only used within entity browsers, but it was inexplicably also the display mode used within the `/browse` maps. I switched the maps to use the "grid_browse" display mode, but this might negatively impact the maps in the short term, since I think you just configured maps to display better. Do you think this is okay, or should we make a new "map_browse" display mode to match "grid_browse" and "list_browse"?

Hey @quicksketch, thanks for explaining this. I double checked, and it should be fine to remove "content_browser". Not sure why `/browse` maps are using that display; it should be "Map Browse".

tl;dr as long as `/admin/structure/views/view/mukurtu_browse` "Map View Block"'s "Rendered Entity Format" is set to "Map Browse" for each item, the display should be good.

quicksketch commented:
> I'm not running into any specific issues, but do have some questions about possible adjustments. I'm happy to merge this PR as is and address those in a follow-up though!

Sure! These changes seem pretty safe, despite how many there are; the previous configuration was fairly consistent and that made replacing easier. I actually changed a whole bunch of the collapsible elements to be open by default, which I thought was an improvement, but then rolled back those changes to ask first.

For example the "Relations" tab, I think the details should be open:

![image](https://github.com/user-attachments/assets/b7b27150-1ee3-4eeb-a829-33f7dd6efadd)

Anyway, let me know what you want and I'll adjust!

michael-wynne-wsu commented:
Alrighty! I know some of this might be more complex or not suitable for this, so we can spin them out into dedicated issues ay any point.

**Assorted**
1) /node/add/word_list > related content is currently using a different widget.

2) I think I agree that having the elements expanded by default is fine, mostly. 
    - On /node/add/collection I would keep "related content" collapsed, but the other two can be expanded
    - On /node/add/person I want to re-order them to "representative terms" > "related people" > "related content". And "related content" can be collapsed, with the others expanded.
    - On /node/add/person "representative terms" is working as intended - this is a regression. It should be pulling taxonomy terms from selections made at /admin/config/mukurtu/taxonomy/records), with a list of recommended terms based on the Name field.

**In the selection widget**

1) How feasible is it to add more filters in addition to type and title? I'm not positive if I want more yet, just trying to think ahead to user requests. 

2) This might just be a me issue, but I immediately went to click on the content names instead of using the checkbox to select them. Maybe a click on row to select would be a bit easier for users?

3) Feasibility of displaying thumbnail preview list? I know that might make the rows too big to be convenient, maybe toggle-able?

4) Feasibility of displaying other fields in preview list? Immediately I'm thinking communities, but like in point 1, TBD.

5) Ideally we would PREVENT self-reference or selecting content more than once. If you re-select an item, nothing happens which is okay, but if you select the active node, it display an error but also cannot be removed.

**In the display widget**

1) Selected content should be order-able.

2) Feasibility of displaying thumbnail previews? (again, I know if might make for bad UI, just want to consider it)

quicksketch commented:
> 1. How feasible is it to add more filters in addition to type and title? I'm not positive if I want more yet, just trying to think ahead to user requests.

Quite easy.

> 2. This might just be a me issue, but I immediately went to click on the content names instead of using the checkbox to select them. Maybe a click on row to select would be a bit easier for users?

I'll see if this is possible. This may need to be split out.

> 3. Feasibility of displaying thumbnail preview list? I know that might make the rows too big to be convenient, maybe toggle-able?

I'll look into this. I don't think Views has an easy way of toggling/collapsing columns but I'll check on adding the "Representative Media" thumbnail.

> 4. Feasibility of displaying other fields in preview list? Immediately I'm thinking communities, but like in point 1, TBD.

Yep, no problem here either except the table might get wide.

> 5. Ideally we would PREVENT self-reference or selecting content more than once. If you re-select an item, nothing happens which is okay, but if you select the active node, it display an error but also cannot be removed.

I'll check. In my testing I did not get an error, but I could also remove the item if I self-referenced. What field was this on?

> 1. Selected content should be order-able.

This should already be the case, you can drag and drop them to order. I tested on Related Content on Dictionary words. Maybe there is a different configuration on one of the fields?

> 2. Feasibility of displaying thumbnail previews? (again, I know if might make for bad UI, just want to consider it)

I think this is fairly easy.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

quicksketch commented:
> /node/add/word_list > related content is currently using a different widget.

Fixed to use a modal browser instead of autocomplete.

>     On /node/add/collection I would keep "related content" collapsed, but the other two can be expanded

Changes made.

>     On /node/add/person I want to re-order them to "representative terms" > "related people" > "related content". And "related content" can be collapsed, with the others expanded.

Changes made.

>     On /node/add/person "representative terms" is working as intended - this is a regression. It should be pulling taxonomy terms from selections made at /admin/config/mukurtu/taxonomy/records), with a list of recommended terms based on the Name field.

Oops, sorry I changed the browser there and shouldn't have. I set it back to the previous browser behavior.

---

Let me know if you want any other changes to references as part of this PR.

quicksketch commented:
> This might just be a me issue, but I immediately went to click on the content names instead of using the checkbox to select them. Maybe a click on row to select would be a bit easier for users?

I could not find a way to do this without custom code. I think the additions would be minimal but this would be better as a separate issue.

michael-wynne-wsu commented:
I changed my initial set of comments to a task list for simplicity and updated that (I added a note about expanded fields on word lists, words, and DH items). I dropped all the browser changes and self-reference into another issue.

Nothing else to add though!

quicksketch commented:
Perfect, thank you I'll make the changes for the last unchecked items!

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

quicksketch commented:
I updated the Collapsed state to be Open on the fields requested.

> Feasibility of displaying thumbnail previews? (again, I know if might make for bad UI, just want to consider it)

Maybe we bundle this one into https://github.com/MukurtuCMS/Mukurtu-CMS/issues/775 too? Both the modal browser and the selected items will have potentially similar problems showing the thumbnail in the admin theme.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr774-w2nxphrxysmcdjr7f67as6xyilmv2zfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e6085006460d7881b532f1

michael-wynne-wsu commented:
@quicksketch agreed about adding the other thumbnail note to the separate issue. I'm very happy to have this merged anytime now!
